### PR TITLE
added test cases for login,logout,signup

### PR DESCRIPTION
**Pull request by prinzpiuz**
added test cases for login,logout,signup

### CODE REVIEW

File: homepage/tests.py [hunk 1]
```diff
@@ -88,3 +89,42 @@ def test_multiple_para_rendering_and_NO_unsafe_tags(self):
...
+            'username': 'testuser1',
+            'password1': 'secret.811',
+            'password2': 'secret.811'}
+        self.input2 = {
+            'username': 'testuser3',
+            'password': 'secret'}
+        User.objects.create_user(**self.input)
+
+    def test_login(self):
+        response = self.client.post(reverse('homepage:login'), self.input, follow=True)
```
Comment (sudheerkdty) [inline homepage/tests.py]: Django's TestClient has a login method which returns True if the user was successfully logged in.
https://docs.djangoproject.com/en/stable/topics/testing/tools/#django.test.Client.login

File: homepage/tests.py [hunk 2]
```diff
@@ -88,3 +89,42 @@ def test_multiple_para_rendering_and_NO_unsafe_tags(self):
...
+
+    def test_logout(self):
+        self.client.post(reverse('homepage:login'), self.input, follow=True)
+        responce = self.client.get(reverse('homepage:logout'))
+        self.assertRedirects(responce, reverse('homepage:login'), status_code=302,
+                             target_status_code=200, fetch_redirect_response=True)
+
+    def test_register(self):
+        responce = self.client.post(reverse('homepage:register'), self.reg, follow=True)
+        self.assertRedirects(responce, reverse('homepage:login'), status_code=302,
```
Comment (zudheer) [inline homepage/tests.py:130]: Also test login for the registered user too.

File: homepage/tests.py [hunk 3]
```diff
@@ -88,3 +89,42 @@ def test_multiple_para_rendering_and_NO_unsafe_tags(self):
...
+                             target_status_code=200, fetch_redirect_response=True)
+
+    # this will test for uniqness of username if tried to login with same username again page dont redirect
+    def test_for_unique_username(self):
+        responce = self.client.post(reverse('homepage:register'), self.reg, follow=True)
+        self.assertEqual(responce.status_code, 200)
```
Comment (zudheer) [inline homepage/tests.py]: check if the response contains error message

File: homepage/tests.py [hunk 4]
```diff
@@ -88,3 +88,62 @@ def test_multiple_para_rendering_and_NO_unsafe_tags(self):
...
+            'password2': 'secret'}
+        self.reg = {
+            'username': 'testuser1',
+            'password1': 'secret.811',
+            'password2': 'secret.811'}
+        self.input2 = {
+            'username': 'testuser3',
+            'password': 'secret'}
+        self.post = {'post_tittle': 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+                      'post_content': 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa<script>alert("Hello");</script>\n\n this is for test'}
```
Comment (sudheerkdty) [inline homepage/tests.py]: self.post and self.input2 is not used anywhere?

Comment (sudheerkdty) [reply homepage/tests.py]: If not remove that

**Pull request merged**